new_symbolic.jl

```diff
@@ -0,0 +1,163 @@
+import YAML
+import DataStructures:OrderedDict
+
+
+function construct_type_map(t::Symbol, constructor::YAML.Constructor,
+                            node::YAML.Node)
+    mapping = _symbol_dict(YAML.construct_mapping(constructor, node))
+    mapping[:tag] = t
+    mapping
+end
+
+
+pairs = [("!Cartesian", :Cartesian),
+         ("!Smolyak", :Smolyak),
+         ("!Normal", :Normal),
+         ("!MarkovChain", :MarkovChain),
+         ("!AR1", :AR1)]
+yaml_types = Dict{AbstractString,Function}([(t, (c, n) -> construct_type_map(s, c, n))
+                       for (t, s) in pairs])
+
+abstract type ASModel end
```

Comment:
`ASModel` is defined as and abstract type here. It is also defined in the Dolo.jl file